nilearn/load_confounds/parser.py

```diff
@@ -0,0 +1,397 @@
+        strategy.append("non_steady_state")
+    return strategy
+
+
+def _check_error(missing_confounds, missing_keys):
+    """Consolidate a single error message across multiple missing confounds."""
+    if missing_confounds or missing_keys:
+        error_msg = (
+            "The following keys or parameters are missing: "
+            + f" {missing_confounds}"
+            + f" {missing_keys}"
+            + ". You may want to try a different denoising strategy."
+        )
+        raise ValueError(error_msg)
+
+
+class Confounds:
+    """
+    Use confounds from fmriprep
+
+    To enable easy confound variables loading from fMRIprep outputs, Confounds
+    provides a interface that groups subsets of confound variables into noise
+    components and their parameters. It is possible to fine-tune a subset of
+    noise components and their parameters through Confounds.
+
+    The implementation will only support fMRIPrep output from the 1.2.x series.
+    The `compcor` noise component requires 1.4.x series or above.
+
+    Parameters
+    ----------
+    strategy : list of strings
+        The type of noise components to include.
+        "motion" head motion estimates.
+        "high_pass" discrete cosines covering low frequencies.
+        "wm_csf" confounds derived from white matter and cerebrospinal fluid.
+        "global" confounds derived from the global signal.
+        "compcor" confounds derived from CompCor.
+        "ica_aroma" confounds derived from ICA-AROMA.
+        "scrub" regressors for Power 2014 scrubbing approach.
+
+    motion : string, optional
+        Type of confounds extracted from head motion estimates.
+        "basic" translation/rotation (6 parameters)
+        "power2" translation/rotation + quadratic terms (12 parameters)
+        "derivatives" translation/rotation + derivatives (12 parameters)
+        "full" translation/rotation + derivatives + quadratic terms + power2d
+               derivatives (24 parameters)
+
+    n_motion_components_components : float
+        Number of pca components to keep from head motion estimates.
+        If the parameters is strictly comprised between 0 and 1, a principal
+        component analysis is applied to the motion parameters, and the number
+        of extracted components is set to exceed `n_motion_components` percent
+        of the parameters variance.
+        If the n_motion_components_components = 0, then no PCA is performed.
+
+    fd_threshold : float, optional
+        Framewise displacement threshold for scrub (default = 0.2 mm)
+
+    std_dvars_threshold : float, optional
+        Standardized DVARS threshold for scrub (default = 3).
+        DVARs is defined as root mean squared intensity difference of volume N
+        to volume N+1 (Power et al. 2012). D referring to temporal derivative
+        of timecourses, VARS referring to root mean squared variance over
+        voxels.
+
+    wm_csf : string, optional
```

Comment:
For these parameters that can only take a finite number of values, I would use brackets with default value first in the list:

```
wm_csf : {"basic", "power2", "derivatives", ....}, optional
```